src/sequence_state.cc

```diff
@@ -226,55 +288,62 @@ SequenceStates::OutputState(
   auto input_states_itr = input_states_.begin();
   std::advance(input_states_itr, iter_advance);
   auto& input_state_r = input_states_[input_states_itr->first];
+  bool reuse_buffer = output_states_[name]->ReuseBuffer();
```

Comment:
in some places we say "use single buffer" in other places we use 'reuse_buffer"
could we normalize the boolean name?